Src/Coravel/Queuing/Queue.cs

```diff
@@ -47,10 +47,16 @@ public Guid QueueTask(Action task)
 
         public Guid QueueInvocableWithPayload<T, TParams>(TParams payload) where T : IInvocable, IInvocableWithPayload<TParams>
         {
+            var tokenSource = new CancellationTokenSource();
             var job = this.EnqueueInvocable<T>(invocable => {
-                IInvocableWithPayload<TParams> invocableWithParams = (IInvocableWithPayload<TParams>) invocable;
+                var invocableWithParams = (IInvocableWithPayload<TParams>) invocable;
                 invocableWithParams.Payload = payload;
+                if (invocableWithParams is ICancellableTask)
```

Comment:
We could also do 
```csharp
if (invocableWithParams is ICancellableTask task)
{
   task.Token = tokenSource.Token;
}
```

Off the top of my head, I think that would work. Would look a bit cleaner? What do you think?

Reply:
Thanks :D commited !!